pkg/controller/goldmane/controller.go

```diff
@@ -144,13 +146,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	reqLogger.Info("Reconciling Goldmane")
 
+	goldmaneDeployment := &v1.Deployment{ObjectMeta: metav1.ObjectMeta{Namespace: common.CalicoNamespace, Name: goldmane.GoldmaneDeploymentName}}
```

Comment:
It would be nice to encapsulate this to make it a bit more clear.

```go
func (r *Reconciler) maintainFinalizer(ctx context.Context, gm client.Object) error {
  // These objects require graceful termination before the CNI plugin is torn down.
  goldmaneDeployment := &v1.Deployment{ObjectMeta: metav1.ObjectMeta{Namespace: common.CalicoNamespace, Name: goldmane.GoldmaneDeploymentName}}
  return utils.MaintainInstallationFinalizer(ctx, r.cli, gm, goldmaneDeployment, render.GoldmaneFinalizer)
```

This just keeps all the logic related to finalizing tidy and in one place.